src/base/file/file_posix.cc

```diff
@@ -69,29 +96,19 @@ bool File::IsExecutable(const String& path, String* error) {
 }
 
 // static
-bool File::Exists(const String& path, String* error) {
-  if (access(path.c_str(), F_OK)) {
-    GetLastError(error);
-    return false;
-  }
-
-  if (!IsFile(path, error)) {
+bool File::Exists(const Path& path, String* error) {
+  std::experimental::filesystem::file_status status;
+  if (!GetStatus(path, &status, error)) {
     return false;
   }
-
-  return true;
+  return std::experimental::filesystem::exists(status) &&
```

Comment:
We anyway need a `status()` inside `IsFile` function, so looks like not a big deal.
Regression only in case when `access` return non-zero which should be rare.